providers/base/bin/prime_offload_tester.py

```diff
@@ -0,0 +1,317 @@
+#!/usr/bin/env python3
+"""prime offload test module."""
+
+import sys
+import threading
+import subprocess
+import time
+import re
+import json
+import argparse
+import logging
+from enum import IntEnum
+
+
+class PrimeOffloaderError(IntEnum):
+    """
+    A class used to define PrimeOffloader Error code
+
+    Attributes
+    ----------
+    NO_ERROR : int
+        process success
+    NO_CARD_ID : int
+        couldn't find card id
+    NO_CARD_NAME : int
+        couldn't find card name
+    OFFLOAD_FAIL : int
+        couldn't find process on specific GPU
+    NOT_SUPPORT_NV_PRIME : int
+        this system couldn't support nv prime offload
+    NV_DRIVER_ERROR : int
+        some errors form nvidia driver
+    """
+
+    NO_ERROR = 0
+    NO_CARD_ID = -1
+    NO_CARD_NAME = -2
+    OFFLOAD_FAIL = -3
+    NOT_SUPPORT_NV_PRIME = -4
+    NV_DRIVER_ERROR = -5
+
+
+class PrimeOffloader:
+    """
+    A class used to execute process to specific GPU.
+    Have to run this as root.
+
+    Attributes
+    ----------
+    logger : obj
+        console logger
+    check_result : int
+        store the result of checking offloading is ok or not.
+
+    Methods
+    -------
+    find_card_id(pci_name)
+        find card id by pci name
+
+    find_card_name(pci_name)
+        find card name by pci name
+
+    check_offload(cmd, card_id, card_name)
+        check offload status
+
+    check_nv_offload_env()
+        check the environment is ok for prime offload
+
+    run_offload_cmd(cmd, pci_name, driver)
+        exectue the command that you would like to offload to specific GPU
```

Comment:
I'm not sure if this part of the docstring is required. As long as you provide a good description docstring for each of the functions below, you don't need to repeat yourself in the class docstring.